tests/integration/issues/2810_test.go

```diff
@@ -0,0 +1,315 @@
+// Copyright 2024 Democratized Data Foundation
```

Comment:
Praise: Thanks a bunch for documenting these! Really nice to see the issue narrowed down with the edge cases you highlighted.